cmd/process-agent/app/status.go

```diff
@@ -0,0 +1,289 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package app
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"text/template"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	"github.com/DataDog/datadog-agent/cmd/process-agent/api"
+	"github.com/DataDog/datadog-agent/pkg/api/util"
+	ddconfig "github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/metadata/host"
+	"github.com/DataDog/datadog-agent/pkg/process/config"
+	ddstatus "github.com/DataDog/datadog-agent/pkg/status"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+var httpClient = util.GetClient(false)
+
+const (
+	statusTemplate = `
+==============================
+Process Agent ({{ .Core.AgentVersion }})
+==============================
+
+ Status date: {{ formatUnixTime .Date }}
+ Process Agent Start: {{ formatUnixTime .Expvars.UptimeNano }}
+ Pid: {{ .Expvars.Pid }}
+ Go Version: {{ .Core.GoVersion }}
+ Python Version: {{ .Core.PythonVersion }}
+ Build arch: {{ .Core.Arch }}
+ Log Level: {{ .Core.Config.LogLevel }}
+ Enabled Checks: {{ .Expvars.EnabledChecks }}
+ Allocated Memory: {{ .Expvars.MemStats.Alloc }} bytes
+ Hostname: {{ .Core.Metadata.Meta.Hostname }}
+
+=================
+Process Endpoints
+=================
+{{- with .Expvars.Endpoints}}
+  {{- range $key, $value := .}}
+  {{$key}} - API Key{{ if gt (len $value) 1}}s{{end}} ending with:
+    {{- range $idx, $apikey := $value }}
+      - {{$apikey}}
```

Comment:
Are we printing the whole API Key or only the last 5 digits?

Reply:
Only the last 5 digits 😊

We truncate it at the expvar level so we don't send api keys over the network